### ISSUE TITLE
Automatically wait for the app to idle

### ISSUE THREAD

**Issue by JoeSSS** [calabash/calabash-ios#1056]
It would be great to have something in Calabash that could detect whether application is in idle state or not. This would help to catch problems with data-fetching and other background processes like crazy animation in background that could provoke battery drain. See it also like a nice improving for wait_for_non_animating.

Some tools could already do that. Tried Xcode7 automation and EarlGrey.

**XCUITest ** implementation is working fine but the implementation is not really convenient because you could only apply it for all project and if on some screen you don't want it you'll need to disable it for whole project.

In **EarlGrey** documentation states: 

> EarlGrey automatically waits for the app to idle by tracking the main dispatch queue, main operation queue, network, animations and several other signals and performs interactions only when the app is Idle

Source: https://github.com/google/EarlGrey/blob/master/docs/api.md

When tried out EarlGrey, I found it interesting and even found a bug in the APP. But implementation of this function is a bit buggy. It somehow relays on viewDidAppear, it should have [super viewDidAppear:animated] for waiting of idle to be working.

**Comment by jmoody** [comment]
Thanks for this suggestion.

@sapieneptus Any ideas how we might achieve the same thing in DeviceAgent?

**Comment by jmoody** [comment]
@JoeSSS Have you seen `wait_for_no_network_indicator`?  That might be interesting...

**Comment by JoeSSS** [comment]
@jmoody Yes I use `wait_for_no_network_indicator ` very widely but this not helps when internal processes are going crazy. I mean, we had some issues with battery drain when CPU load was 107% but it was nothing to do with network indicator.

**Comment by sapieneptus** [comment]
@jmoody It's a good question, I'm not 100% sure if we can get that info out of process but it's worth investigating.

**Comment by ark-konopacki** [comment]
@jmoody only guessing can we get a cpu load?
